cmd/kogito/command/deploy/deploy_service.go

```diff
@@ -157,9 +166,34 @@ func (i *deployCommand) Exec(cmd *cobra.Command, args []string) (err error) {
 	log := context.GetDefaultLogger()
 	i.flags.name = args[0]
 	hasSource := false
+	options := &buildv1.BinaryBuildRequestOptions{}
+	var userProvidedFile io.Reader
+
 	if len(args) > 1 {
-		i.flags.source = args[1]
-		hasSource = true
+		userProvidedFile, options, err = matchKogitoResources(args[1])
+		if err != nil {
+			switch t := err.(type) {
+			case *os.PathError:
+				return t
+
+			case *kogitoerror.KogitoAssetNotFound:
+				return t
+
+			case *url.Error:
+				return t
+
+			default:
+				fmt.Println(reflect.TypeOf(t))
+				log.Warnf(message.KogitoAppAssetNotSupported)
```

Comment:
It would be good to add the asset path as part of warning message.

Reply:
good point.